instrumentation/twilio-6.6/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/twilio/TwilioClientTest.java

```diff
@@ -0,0 +1,546 @@
+          + "      \"price\": -0.03000,\n"
+          + "      \"price_unit\": \"USD\",\n"
+          + "      \"sid\": \"CAXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\",\n"
+          + "      \"start_time\": \"Tue, 31 Aug 2010 20:36:29 +0000\",\n"
+          + "      \"status\": \"completed\",\n"
+          + "      \"subresource_uris\": {\n"
+          + "        \"notifications\": \"/2010-04-01/Accounts/ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/Calls/CAXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/Notifications.json\",\n"
+          + "        \"recordings\": \"/2010-04-01/Accounts/ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/Calls/CAXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/Recordings.json\",\n"
+          + "        \"feedback\": \"/2010-04-01/Accounts/ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/Calls/CAXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/Feedback.json\",\n"
+          + "        \"feedback_summaries\": \"/2010-04-01/Accounts/ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/Calls/FeedbackSummary.json\"\n"
+          + "      },\n"
+          + "      \"to\": \"+15558675310\",\n"
+          + "      \"to_formatted\": \"(555) 867-5310\",\n"
+          + "      \"uri\": \"/2010-04-01/Accounts/ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/Calls/CAXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX.json\"\n"
+          + "    }";
+  private static final String ERROR_RESPONSE_BODY =
+      "{\n"
+          + "      \"code\": 123,\n"
+          + "      \"message\": \"Testing Failure\",\n"
+          + "      \"code\": 567,\n"
+          + "      \"more_info\": \"Testing\"\n"
+          + "    }";
+
+  private TwilioRestClient twilioRestClient;
+
+  @BeforeAll
+  static void setUp() {
+    Twilio.init(ACCOUNT_SID, AUTH_TOKEN);
+  }
+
+  @AfterAll
+  static void tearDown() {
+    Twilio.getExecutorService().shutdown();
+    Twilio.setExecutorService(null);
+    Twilio.setRestClient(null);
+  }
+
+  @Test
+  void synchronousMessage() {
+    twilioRestClient = mock(TwilioRestClient.class);
+    when(twilioRestClient.getObjectMapper()).thenReturn(new ObjectMapper());
+    when(twilioRestClient.request(any()))
+        .thenReturn(
+            new Response(
+                new ByteArrayInputStream(MESSAGE_RESPONSE_BODY.getBytes(StandardCharsets.UTF_8)),
+                200));
+
+    Message message =
+        testing.runWithSpan(
+            "test",
+            () ->
+                Message.creator(
+                        new PhoneNumber("+1 555 720 5913"),
+                        new PhoneNumber("+1 555 555 5215"),
+                        "Hello world!")
+                    .create(twilioRestClient));
+
+    assertThat(message.getBody()).isEqualTo("Hello, World!");
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("test").hasNoParent().hasAttributes(Attributes.empty()),
+                span ->
+                    span.hasName("MessageCreator.create")
+                        .hasKind(CLIENT)
```

Comment:
usually we also add `.hasParent(trace.getSpan(0))`

Reply:
done